pandas/tests/dtypes/test_dtypes.py

```diff
@@ -51,6 +51,36 @@ def test_pickle(self):
         assert result == self.dtype
 
 
+class TestBaseDtype:
```

Comment:
Such tests are already in the base extension tests (tests/extension/base/dtype), or can be expanded there

Reply:
I had the feeling that those were only base tests for implementing tests for custom extensions, couldn't find a place where I thought it'd make sense to have the abstract class test. Where exactly would you have them?

Reply:
>  those were only base tests for implementing tests for custom extensions

Yes, but we have custom extensions ourselves, so we use those tests ourselves (and in addition we also have a few extra test EAs). 

For example in `tests/extension/decimal/array.py` you can remove the custom implementation and let it inherit, and then it is already tested in that way.

> where I thought it'd make sense to have the abstract class test

What do you mean with abstract class test?

Reply:
I don't think I understand what you want to test.

@jreback wanted a test for the existence of the method `construct_from_string`, which makes sense to me. Ideally, we'd already have a test, testing that the method raises an exception when called directly from `ExtensionDtype` (what I called the abstract class before). Then I'd just replace that by checking that now it's implemented, and it can construct an instance given the string of the name of the dtype.

I saw that in `extension/base/dtype.py` we already have a test for `construct_from_string`, but that's meant for custom extensions (ours or from a third-party) to test that their implementation of `construct_from_string` follows the expected behavior. The way of using that test is to create a subclass of `BaseDtypeTests`.

I don't think we want to test that the method is implemented in the `ExtensionDtype` (the base/abstract class) by checking a subclass of it. So, I see two options:

- Create a subclass of `BaseDtypeTests` defining that the `dtype` to test is not a subclass but `ExtensionDtype` itself (it will fail for the abstract methods that still exist in `ExtensionDtype).
- Create an independent test for `ExtensionDtype` and not their subclasses, that doesn't subclass `BaseDtypeTests` (this is what I've done here, not testing only `construct_from_string` but all the methods that I could tests, since there weren't tests for the the "default"/base implementations).

Am I missing something here? Sorry I don't understand exactly what you propose.

Reply:
see @jorisvandenbossche comments on where tests should go, we have a very general framework for these already.